modules/mongodb/mongodb_test.go

```diff
@@ -14,32 +14,52 @@ import (
 
 func TestMongoDB(t *testing.T) {
 	type tests struct {
-		name  string
-		image string
+		name string
+		opts []testcontainers.ContainerCustomizer
 	}
 	testCases := []tests{
 		{
-			name:  "From Docker Hub",
-			image: "mongo:6",
+			name: "From Docker Hub",
+			opts: []testcontainers.ContainerCustomizer{
+				testcontainers.WithImage("mongo:6"),
+			},
 		},
 		{
-			name:  "Community Server",
-			image: "mongodb/mongodb-community-server:7.0.2-ubi8",
+			name: "Community Server",
+			opts: []testcontainers.ContainerCustomizer{
+				testcontainers.WithImage("mongodb/mongodb-community-server:7.0.2-ubi8"),
+			},
 		},
 		{
-			name:  "Enterprise Server",
-			image: "mongodb/mongodb-enterprise-server:7.0.0-ubi8",
+			name: "Enterprise Server",
+			opts: []testcontainers.ContainerCustomizer{
+				testcontainers.WithImage("mongodb/mongodb-enterprise-server:7.0.0-ubi8"),
+			},
+		},
+		{
+			name: "With Replica set and mongo:4",
+			opts: []testcontainers.ContainerCustomizer{
+				testcontainers.WithImage("mongo:4"),
+				mongodb.WithReplicaSet(),
+			},
+		},
+		{
+			name: "With Replica set and mongo:6",
+			opts: []testcontainers.ContainerCustomizer{
+				testcontainers.WithImage("mongo:6"),
+				mongodb.WithReplicaSet(),
+			},
 		},
 	}
 
-	for _, tc := range testCases {
-		image := tc.image
-		t.Run(image, func(t *testing.T) {
+	for _, tt := range testCases {
```

Comment:
BTW let's keep this as `tc`. Can you rename the inner `t` instance as `tt`? Then please call `tt.Parallel` in L58 🙏 that will avoid possible confusions around what instance calls the parallel

Reply:
Sure, I also added `opts := tc.opts` outside the `t.Run` function to prevent the 'loop variable tc captured by func literal' on L61.

Reply:
I ended up not reading the last line, sorry. I converted `opts := tc.opts` to `tc := tc`.